frontend/src/app/components/network-graph/network-graph.component.ts

```diff
@@ -821,8 +832,38 @@ export class NetworkGraphComponent implements OnInit, OnChanges {
     this.infoBoxOverlay.selectAll('.info').remove();
   }
 
-  private selectNode(selectedNode: Node) {
+  private selectNode(selectedNode: SimulationNode) {
+    this.highlightNode(selectedNode);
+    this.highlightedNode = selectedNode;
```

Comment:
Is there ever a chance this two are not the same? What I mean: can I set the highlighted node without highlighting it or the other way around? Btw. why are there the two terminologies of selected versus highlighted node? They seem to be used as the same (according to the function name and the code below). Or am I wrong. What I would understand is that there is a selected node which then gets highlighted. But then it is still the selected node just with a visual effect. Or it is just the highlighted node in general without any selection terminology. :shrug: 
But maybe I'm totally of. Sorry for that. :see_no_evil:

Reply:
These terminologies are used as the same. You can't select a node without highlighting it. I will improve the naming here a bit. So there will be a selected node that gets highlighted once and for that will be the info box drawn every tick.